e2e/tests/account-settings.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { test } from '../fixtures';
+import { expect } from 'playwright/test';
+import { USER_EMAIL, USER_PASSWORD } from '../constants';
+import { getPassword } from '../fake-data-generation';
+
+test.describe('Account settings', () => {
+  test.beforeEach('Sign in', async ({ signInPage }) => {
+    if (!USER_EMAIL) {
+      test.skip(!USER_EMAIL, 'PLAYWRIGHT_USER_EMAIL was not defined');
+      return;
+    }
+    if (!USER_PASSWORD) {
+      test.skip(!USER_PASSWORD, 'PLAYWRIGHT_USER_PASSWORD was not defined');
+      return;
+    }
+
+    await signInPage.signIn({ email: USER_EMAIL, password: USER_PASSWORD });
+  });
+
+  test('Verify navigation to the Account page via account nav icon', async ({ accountPage }) => {
+    await accountPage.openAccountSettings();
+  });
+
+  test('Sign Out', async ({ accountPage }) => {
+    await accountPage.navigateToAccountPage();
+    await accountPage.signOut();
+  });
+
+  test.describe('Account / Profile', () => {
+    test.beforeEach('Navigate to the Account settings', async ({ accountPage }) => {
+      await accountPage.navigateToAccountPage();
+    });
+
+    test('Verify red messages appear if required fields are not filled in', async ({ page, accountPage }) => {
+      await accountPage.clearAllRequiredFieldsForProfile();
+      await accountPage.profileSaveBtn().click();
+      await expect(page.getByText('This field is required')).toHaveCount(3);
+    });
+
+    test('Verify user can update personal information', async ({ accountPage }) => {
+      const newFirstName = await accountPage.changeProfileFirstName();
+      const newLastName = await accountPage.changeProfileLastName();
+      const newPhoneNumber = await accountPage.changePhoneNumber();
+
+      await accountPage.saveProfileUpdates();
```

Comment:
What are your thoughts on refreshing after the save to make sure the fields were really saved rather than just still in the form?

Reply:
great idea, done